oneview_redfish_toolkit/api/server_hardware_resource_block.py

```diff
@@ -0,0 +1,106 @@
+# -*- coding: utf-8 -*-
+
+# Copyright (2018) Hewlett Packard Enterprise Development LP
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+from oneview_redfish_toolkit.api.resource_block import ResourceBlock
+
+
+class ServerHardwareResourceBlock(ResourceBlock):
+    """Creates a ResourceBlock Redfish dict for Server Hardware
+
+        Populates self.redfish with some hardcoded ResourceBlock
+        values and with Server Hardware data retrieved from OneView.
+    """
+
+    def __init__(self, uuid, server_hardware, server_profile_templates):
+        """ServerHardwareResourceBlock constructor
+
+            Populates self.redfish with the contents of ServerHardware dict.
+
+            Args:
+                uuid: server hardware UUID
+                server_hardware: ServerHardware dict from OneView
+                server_profile_templates: list of OneView server profile
+                templates
+        """
+        super().__init__(uuid, server_hardware)
+
+        self.server_hardware = server_hardware
+        self.server_profile_templates = server_profile_templates
+
+        self.redfish["ResourceBlockType"] = ["Compute"]
+
+        profile_applied = self.server_hardware["state"] == "ProfileApplied"
```

Comment:
Is there a state returned by OneView for "Composing" and "Failed"?

Reply:
The following status will be added:

- NoProfileApplied (server successfully added) --> Unused
- ApplyingProfile (profile being applied to server) --> Composing
- ProfileApplied (profile successfully applied) --> Composed
- ProfileError (unsuccessful profile apply or removal) --> Failed